File: AlgorithmsPY/selection_sort.py
##
# Selection Sort function O(N^2 / 2)
##
def selection_sort(list):
  list_length = len(list)
  for i in range(list_length):
    lower_number_index = i
    for j in range(i, list_length):
      if list[j] < list[lower_number_index]:
        lower_number_index = j

    if lower_number_index != i:
      list[i], list[lower_number_index] = list[lower_number_index], list[i]

  return list

File: AlgorithmsJS/SelectionSort.js
/**
 * Selection Sort function O(N^2 / 2)
 */
function selectionSort(list) {
  const listLength = list.length;
  for (let i = 0; i < listLength; i++) {
    let lowerNumberIndex = i;
    for (let j = i; j < listLength; j ++) {
      if (list[j] < list[lowerNumberIndex]) {
        lowerNumberIndex = j;
      }
    }

    if (lowerNumberIndex != i) {
      const temp = list[i];
      list[i] = list[lowerNumberIndex];
      list[lowerNumberIndex] = temp;
    }
  }

  return list;
}

module.exports = selectionSort;
